oshdb-api/src/main/java/org/heigit/bigspatialdata/oshdb/api/generic/OSHDBCombinedIndex.java

```diff
@@ -12,16 +12,22 @@ public OSHDBCombinedIndex(U index1, V index2) {
   }
 
   public U getFirstIndex() {
-    return this.payload.getLeft();
+    return this.index1;
   }
 
   public V getSecondIndex() {
-    return this.payload.getRight();
+    return this.index2;
   }
 
   @Override
   public int compareTo(@NotNull OSHDBCombinedIndex<U,V> o) {
-    return this.payload.compareTo(o.payload);
+	final Comparable<Object> ci1 = (Comparable<Object>) index1;
+    int c = ci1.compareTo(o.index1);
+    if(c == 0) {
```

Comment:
missing space between `if` and the `(`: https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace